### PR TITLE
feat: infer the user's package manager for installing dependencies

### LINKED ISSUE #96: Support using package managers other than npm
Hi There 👋 

It would be great to be able to specify which package manager to use when creating an app. Right now, if a user wants to use `pnpm`, they have to manually interrupt the package installation (which is awkward in itself), go to the app folder, delete the `package-lock.json`, and run `pnpm i`.

I'm happy to create a PR if you accept this feature request.

### PR DESCRIPTION
**Pull request by Lehoczky**
## Summary of change

This PR detects the user's package manager with [which-pm-runs](https://github.com/zkochan/packages/tree/main/which-pm-runs), and uses it if a package manager hasn't been explicitly specified with the `--manager` command line argument.

Additionally it turned out that when installing packages, the script did not take into consideration the `--manager` argument, `npm install` was hard-coded at multiple places. It has been fixed as well.

## Related issues

Closes #96 

## Test plan

-   [ ] If added a new boilerplate
    -   I tested the new boilerplate by running the CLI locally
    -   I tested that the boilerplate is created and works correctly when using command line flags (`--recipe=...` for example)
-   [ ] If added a new recipe, frontend or backend
    -   I tested that the newly added option is usable by passing command line flags (`--recipe=...` for example)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [ ] Had run `npm run build-pretty`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] If added a new recipe, I also modified types to include the new recipe in `Recipe` and `allRecipes`
-   [ ] If added a new frontend, I also modified types to include the new frontend in `SupportedFrontends` and `allFrontends` if required
-   [ ] If added a new backend, I also modified types to include the new backend in `SupportedBackends` and `allBackends` if required

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Enforce Unix newlines
+* text=auto eol=lf
```
Comment (Lehoczky) [inline .gitattributes:2]: Windows users need this to not have issues with line endings

**Comment by Lehoczky** [comment]
Pre-commit hooks cannot be installed on Windows with `npm run set-up-hooks`, because the script has unix commands like `cp`.

I would recommend using [husky](https://typicode.github.io/husky/), it installs the hooks automatically and works on every platform.

File: lib/build/config.js [hunk 2]
```diff
@@ -14,8 +14,7 @@ function getCapacitorMessage() {
         return `create-supertokens-app uses ${chalk.bold("nextjs-native-starter")} as a template app for Capacitor. ${chalk.bold("nextjs-native-starter")} is a starter project created by ${chalk.blue("https://github.com/RobSchilderr")}, if you face any problems please open an issue here: ${chalk.blue("https://github.com/RobSchilderr/nextjs-native-starter/issues")} or reach out to us on the SuperTokens Discord server: ${chalk.blue("https://supertokens.com/discord")}.\n\nTo get started refer to the README on Github ${chalk.blue("https://github.com/RobSchilderr/nextjs-native-starter")}`;
     }
 }
-export async function getFrontendOptions(userArguments) {
-    const packagerCommand = await getPackageManagerCommand(userArguments);
```
Comment (Lehoczky) [inline lib/build/config.js:18]: `getPackageManagerCommand` did nothing, so I've removed it

File: lib/ts/utils.ts [hunk 3]
```diff
@@ -168,7 +169,7 @@ function getPackageJsonString(input: {
             "start:backend": "cd backend && ${backendStartScript}",
             "start:backend-live-demo-app": "cd backend && ./startLiveDemoApp.sh",
             "start": "npm-run-all --parallel start:frontend start:backend",
-            "start-live-demo-app": "npx npm-run-all --parallel start:frontend-live-demo-app start:backend-live-demo-app"
+            "start-live-demo-app": "npm-run-all --parallel start:frontend-live-demo-app start:backend-live-demo-app"
```
Comment (rishabhpoddar) [inline lib/ts/utils.ts:172]: this command is for our live demo apps hosted by us, so please undo this change

File: lib/ts/utils.ts [hunk 4]
```diff
@@ -675,7 +676,7 @@ export async function runProjectOrPrintStartCommand(answers: Answers, userArgume
         return;
     }
 
-    let appRunScript = `${await getPackageManagerCommand(userArguments)} run start`;
+    let appRunScript = `${userArguments.manager} run start`;
 
     if (selectedFrontend.isFullStack) {
```
Comment (rishabhpoddar) [inline lib/ts/utils.ts:681]: also need a version change:
- version.ts
- package.json

File: lib/build/config.js [hunk 5]
```diff
@@ -14,8 +14,7 @@ function getCapacitorMessage() {
         return `create-supertokens-app uses ${chalk.bold("nextjs-native-starter")} as a template app for Capacitor. ${chalk.bold("nextjs-native-starter")} is a starter project created by ${chalk.blue("https://github.com/RobSchilderr")}, if you face any problems please open an issue here: ${chalk.blue("https://github.com/RobSchilderr/nextjs-native-starter/issues")} or reach out to us on the SuperTokens Discord server: ${chalk.blue("https://supertokens.com/discord")}.\n\nTo get started refer to the README on Github ${chalk.blue("https://github.com/RobSchilderr/nextjs-native-starter")}`;
     }
 }
```
Comment (rishabhpoddar) [inline lib/build/config.js:16]: one question is that if a user is using `pnpm` for example, the generated demo apps (for node + react) will stilljust use npm. Do you think that's ok?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/build/config.js (hunk 5 above)
Comment (Lehoczky) [reply lib/build/config.js:16]: Do you mean that the scripts in the `package.json` file will be run with `npm`, like `npm run start`? Sure, that's completely okay!

These package managers do not have differences, when running these scripts. They vary in syntax (which is correctly handled here, I checked), but that's all.

File: lib/ts/utils.ts (hunk 3 above)
Comment (Lehoczky-peakfs) [reply lib/ts/utils.ts:172]: I'll gladly revert this change, but these commands are the same. when you run a package.json script with `npm run ...` npm will automatically use `npx` with that command. Writing `npx` here at the start is not necessary.

File: lib/ts/utils.ts (hunk 4 above)
Comment (Lehoczky-peakfs) [reply lib/ts/utils.ts:681]: 🫡 May I add this to the contribution guide to always update the version?

File: lib/build/config.js (hunk 5 above)
Comment (rishabhpoddar) [reply lib/build/config.js:16]: Gotcha!

File: lib/ts/utils.ts (hunk 4 above)
Comment (rishabhpoddar) [reply lib/ts/utils.ts:681]: It's there in the PR checklist template above already. But yea, adding to the contributing guide always helps too :)

File: lib/ts/utils.ts (hunk 3 above)
Comment (rishabhpoddar) [reply lib/ts/utils.ts:172]: oh yea, but i don't think the server in which we run the command installs `npm-run-all` when `npm i` is ran. So it will fail without `npx` in front. I don't recall entirely, so prefer to just keep this the way it is

Comment (Lehoczky) [reply lib/ts/utils.ts]: Done!

File: lib/ts/utils.ts (hunk 4 above)
Comment (Lehoczky) [reply lib/ts/utils.ts:681]: Aaand done!

**Comment by rishabhpoddar** [comment]
Thanks @Lehoczky !
